bugsnag_test.go

```diff
@@ -23,15 +23,20 @@ type exception struct {
 	Message    string       `json:"message"`
 	Stacktrace []stackFrame `json:"stacktrace"`
 }
+
+type event struct {
+	Exceptions []exception      `json:"exceptions"`
+	Metadata   bugsnag.MetaData `json:"metaData"`
+}
+
 type notice struct {
-	Events []struct {
-		Exceptions []exception `json:"exceptions"`
-	} `json:"events"`
+	Events []event `json:"events"`
 }
 
 func TestNoticeReceived(t *testing.T) {
```

Comment:
If we decide to always pass the `metadata` param to `Notify`, I don't really see the need to test the case where `metadata` is empty. I'd just have one test that also validates the metadata values.